builder/builder_test.go

```diff
@@ -307,3 +358,178 @@ func getAppHashFromEVM(ethState *state.StateDB, header *monomer.Header) (common.
+	subscription := subscribeToEventBus(t, eventBus, subChannelLen)
+
+	generatedBlock := testutils.GenerateBlock(t)
+
+	// Get the L1 info & deposit transactions (combined)
+	depositTxs := generatedBlock.Txs[0]
+
+	ethTxs, err := monomer.AdaptCosmosTxsToEthTxs(bfttypes.Txs{depositTxs})
+	require.NoError(t, err)
+	require.Len(t, ethTxs, 2, "Expected two Ethereum transactions: L1 attribute and deposit tx, got %d", len(ethTxs))
+
+	depositTxETH := ethTxs[1]
+	require.NotNil(t, depositTxETH.Value())
+	require.NotNil(t, depositTxETH.To(), "Deposit transaction must have a 'to' address")
+
+	cosmAddr := monomer.EvmToCosmosAddress(*depositTxETH.To())
+	withdrawalTx := testapp.ToWithdrawalTx(
+		t,
+		cosmAddr.String(),
+		common.HexToAddress("0x12345abcde").String(),
+		math.NewIntFromBigInt(depositTxETH.Value()),
+	)
+
+	b := builder.New(
+		pool,
+		app,
+		blockStore,
+		txStore,
+		eventBus,
+		g.ChainID,
+		ethstatedb,
+	)
+
+	// Prepare the payload
+	txs := bfttypes.Txs{depositTxs, withdrawalTx}
+	payload := &builder.Payload{
+		InjectedTransactions: txs,
+		GasLimit:             1000000000000,
+		Timestamp:            g.Time + 1,
+		NoTxPool:             true,
+	}
+
+	// Build a block with the payload
+	builtBlock, preBuildInfo, postBuildInfo := buildBlock(t, b, app, payload)
+
+	// Test deposit was received
+	checkDepositTxResult(t, txStore, depositTxs, fmt.Sprintf("%sETH", depositTxETH.Value().String()), cosmAddr.String())
+
+	withdrawalTxResult, err := txStore.Get(bfttypes.Tx(withdrawalTx).Hash())
+	require.NoError(t, err)
+	require.NotNil(t, withdrawalTxResult)
+	require.Truef(t, withdrawalTxResult.Result.IsOK(), "Expected the withdrawal transaction to be successful, but it failed")
+
+	// Verify block creation
+	genesisBlock, err := blockStore.BlockByHeight(uint64(preBuildInfo.GetLastBlockHeight()))
+	require.NoError(t, err)
+	gotBlock, err := blockStore.HeadBlock()
+	require.NoError(t, err)
+
+	ethStateRoot := gotBlock.Header.StateRoot
+	header := monomer.Header{
+		ChainID:    g.ChainID,
+		Height:     uint64(postBuildInfo.GetLastBlockHeight()),
+		Time:       payload.Timestamp,
+		ParentHash: genesisBlock.Header.Hash,
+		StateRoot:  ethStateRoot,
+		GasLimit:   payload.GasLimit,
+	}
+	wantBlock, err := monomer.MakeBlock(&header, txs)
+	require.NoError(t, err)
+	verifyBlockContent(t, wantBlock, gotBlock, builtBlock)
+
+	// We expect two transactions: one combined transaction (l1InfoTx + depositTxs) and one withdrawal transaction (withdrawalTx).
+	require.Equal(t, 2, builtBlock.Txs.Len(), "Expected the built block to contain 2 transactions: depositTxs and withdrawalTx")
+
+	// Test parseWithdrawalMessages
+	checkWithdrawalTxResult(t, withdrawalTxResult)
+
+	expectedStateRoot := wantBlock.Header.StateRoot
+	gotStateRoot := gotBlock.Header.StateRoot
+	require.Equal(t, expectedStateRoot, gotStateRoot, "Expected the built block to contain state root hash")
+
+	for i := range gotBlock.Txs {
+		select {
+		case event, ok := <-subscription.Out():
```

Comment:
Could you extract the `select` statement out into a helper func since it's the same logic that the `TestBuild` test uses to verify the event bus is behaving properly?

Reply:
Done